File: rnmodules/react-native-kb/android/src/main/java/com/reactnativekb/ReadFileAsString.java
package com.reactnativekb;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStreamReader;

public class ReadFileAsString {
  public static String read(String path) {
    String ret = "";

    try {
      FileInputStream inputStream = new FileInputStream(new File(path));

      if (inputStream != null) {
        InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
        BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
        String receiveString = "";
        StringBuilder stringBuilder = new StringBuilder();

        while ((receiveString = bufferedReader.readLine()) != null) {
          stringBuilder.append(receiveString);
        }

        inputStream.close();
        ret = stringBuilder.toString();
      }
    } catch (FileNotFoundException e) {
      // ignore
    } catch (IOException e) {
      // ignore
    }

    return ret;
  }
}

File: rnmodules/react-native-kb/android/src/main/java/com/reactnativekb/ReadFileAsString.kt
package com.reactnativekb

import java.io.File
import java.io.FileNotFoundException
import java.io.IOException

object ReadFileAsString {
    fun read(path: String): String {
        return try {
            File(path).readText()
        } catch (e: FileNotFoundException) {
            ""
        } catch (e: IOException) {
            ""
        }
    }
}
